test_bandwidth.py

```diff
@@ -0,0 +1,204 @@
+import os
+import sys
+import time
+import argparse
+from abc import abstractmethod
+
+import torch
+import torch.distributed as dist
+from typing import List
+from prettytable import PrettyTable
+os.environ['CUDA_VISIBLE_DEVICES'] = '1,2,3,4'
+    
+def setup(rank, world_size):
+    os.environ["MASTER_ADDR"] = os.getenv("MASTER_ADDR", "localhost")
+    os.environ["MASTER_PORT"] = os.getenv("MASTER_PORT", "12355")
+    os.environ["RANK"] = os.getenv("RANK", "0")
+    os.environ["WORLD_SIZE"] = os.getenv("WORLD_SIZE", "1")
+    dist.init_process_group(world_size=world_size, rank=rank,
+                            init_method="env://", backend="nccl")
+    torch.cuda.set_device(rank)
```

Comment:
This is not right, what if the rank is more than 8, i.e. I am running with 16 GPUs.

Reply:
Okay, got it, I'll fix it.